bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/folder/FolderObserver.java

```diff
@@ -238,7 +238,19 @@ public boolean accept(Path entry) throws IOException {
     }
 
     private void checkPath(final Path path, final WatchService.Kind kind) {
+        String fileName = path.getFileName().toString();
```

Comment:
If I am not wrong, you extracted the changes done in that class in a separate PR that was merged.
So you need to adjust that PR.

Reply:
Yes, this has already been merged - and I have extracted another PR from this (#4910) which should be merged by itself, as it's not really a part of this. When that is done, I'll rebase this PR so that these parts disappear. But, I don't want to do the rebasing now, as I'm afraid it will interfere with all your review comments. So, just ignore that part.